pkg/common/finalizer_util.go

```diff
@@ -0,0 +1,41 @@
+package common
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// ContainsFinalizer checks if finalizer is found in metadata
+func ContainsFinalizer(metadata metav1.ObjectMeta, finalizer string) bool {
+	for _, f := range metadata.Finalizers {
+		if f == finalizer {
+			return true
+		}
+	}
+	return false
+}
+
+// SetFinalizer either adds or removes finalizer depending on `setFinalizer`
+// it returns a bool indicating whether the metadata was changed or not.
+func SetFinalizer(metadata *metav1.ObjectMeta, finalizer string, setFinalizer bool) bool {
```

Comment:
Can we have two different functions for this  please?

Reply:
checking

Reply:
Hey, can you explain why you want to spit this? I'd have to rework the other the unit tests too, and It does seem cleaner to do:

```
if (something){
   needsFinalizer = false
else if (something else) { 
   needsFinalizer = true
}

err := setFinalizer(needsFinalizer)
// handle err

```

than

```
if (something){
   err := unSetFinalizer()
   //handle err
else if (something else) { 
   err := SetFinalizer()
  // handle err
}
```